aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.spec.move

```diff
@@ -1,11 +1,56 @@
 spec aptos_std::multi_ed25519 {
-    spec public_key_validate_internal {
-        // TODO: temporary mockup.
+
+    // -----------------------
+    // Function specifications
+    // -----------------------
+
+    spec new_unvalidated_public_key_from_bytes(bytes: vector<u8>): UnvalidatedPublicKey {
+        let length = len(bytes);
+        aborts_if length / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS;
+        aborts_if length % INDIVIDUAL_PUBLIC_KEY_NUM_BYTES != THRESHOLD_SIZE_BYTES;
+        ensures result == UnvalidatedPublicKey { bytes };
+    }
+
+    spec new_validated_public_key_from_bytes(bytes: vector<u8>): Option<ValidatedPublicKey> {
+        aborts_if len(bytes) % INDIVIDUAL_PUBLIC_KEY_NUM_BYTES == THRESHOLD_SIZE_BYTES
+            && len(bytes) / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS;
+        let cond = len(bytes) % INDIVIDUAL_PUBLIC_KEY_NUM_BYTES == THRESHOLD_SIZE_BYTES
+            && spec_public_key_validate_internal(bytes);
+        ensures cond ==> result == option::spec_some(ValidatedPublicKey{bytes});
+        ensures !cond ==> result == option::spec_none<ValidatedPublicKey>();
+    }
+
+    spec new_signature_from_bytes(bytes: vector<u8>): Signature {
+        aborts_if len(bytes) % INDIVIDUAL_SIGNATURE_NUM_BYTES != BITMAP_NUM_OF_BYTES;
+        ensures result == Signature { bytes };
+    }
+
+
+    // ----------------
+    // Native functions
+    // ----------------
+
+    spec fun spec_public_key_validate_internal(bytes: vector<u8>): bool;
+
+    spec fun spec_signature_verify_strict_internal(
+        multisignature: vector<u8>,
+        public_key: vector<u8>,
+        message: vector<u8>
+    ): bool;
+
+    spec public_key_validate_internal(bytes: vector<u8>): bool {
         pragma opaque;
+        aborts_if len(bytes) / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS;
```

Comment:
Hey @junkil-park, I see this says that the native aborts upon this condition, but the native never aborts.
See https://github.com/aptos-labs/aptos-core/blob/82180081deca780ecf988d0be8d780c38321d505/aptos-move/framework/src/natives/cryptography/multi_ed25519.rs#L31

Reply:
I think my previous reply in a discussion above made it seem like native aborts upon this condition, but it merely returns false.

Reply:
Thank you letting me know! I put up the fix here: https://github.com/aptos-labs/aptos-core/pull/5832/files#diff-ed6845718b5d03098f058edfc43ace50fc05ff3d18bf729521701a36b0c19bc3R45